### PR TITLE
Upgrade ADF to Angular 16

### PR DESCRIPTION
**Pull request by swapnil-verma-gl**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [x] Other... Please describe:
Updated angular to v16

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: .github/workflows/release-branch.yml
Comment (DenysVuika) [inline .github/workflows/release-branch.yml:1]: we could probably move those changes to the ADF earlier, to have less files modified

File: .gitignore
Comment (DenysVuika) [inline .gitignore:1]: we could probably move those changes to the ADF earlier, to have less files modified

File: decorate-angular-cli.js
Comment (DenysVuika) [inline decorate-angular-cli.js:1]: this file should be deleted completely from the ADF, we are not using it for ages

File: docs/README.md
Comment (DenysVuika) [inline docs/README.md:1]: changes to this file can be either moved to ADF earlier, or reverted

File: docs/release-notes/RelNote410.md
Comment (DenysVuika) [inline docs/release-notes/RelNote410.md:1]: let's reformat that earlier in ADF to have less changes

File: docs/release-notes/RelNote420.md
Comment (DenysVuika) [inline docs/release-notes/RelNote420.md:1]: let's reformat that earlier in ADF to have less changes

File: docs/release-notes/RelNote430.md
Comment (DenysVuika) [inline docs/release-notes/RelNote430.md:1]: let's reformat that earlier in ADF to have less changes

File: e2e-playwright/page-object/components/material/material-locators.ts [hunk 1]
```diff
@@ -20,6 +20,7 @@ export const materialLocators = {
     root: 'mat-error'
   },
   Tooltip: {
-    root: 'mat-tooltip-component'
+    root: 'mat-tooltip-component',
+    message: '.mdc-tooltip__surface'
```
Comment (DenysVuika) [inline e2e-playwright/page-object/components/material/material-locators.ts:24]: the material tooltips are going away soon in ADF, can this be avoided?

File: e2e-playwright/page-object/components/material/validation.component.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ import { materialLocators } from './material-locators';
 
 export class TooltipComponent extends BaseComponent {
     private static rootElement = materialLocators.Tooltip.root;
-    public content = this.getChild('div');
+    public content = this.getChild(materialLocators.Tooltip.message);
```
Comment (DenysVuika) [inline e2e-playwright/page-object/components/material/validation.component.ts:24]: this whole class is being deprecated in ADF right now

File: e2e-playwright/page-object/stories/base-stories.ts
Comment (DenysVuika) [inline e2e-playwright/page-object/stories/base-stories.ts:1]: can this be done earlier in ADF to avoid changes?

File: e2e/process-services-cloud/form-field/dropdown-widget.e2e.ts [hunk 3]
```diff
@@ -130,7 +130,7 @@ describe('Form Field Component - Dropdown Widget', () => {
         expect(optionsSelected.toString().replace(/\s+/g, '')).toEqual([optionsToSelect, optionsToSelect].toString());
     });
 
-    it('[C309878] Should be able to select a dropdown option, save and complete the task form', async () => {
+    fit('[C309878] Should be able to select a dropdown option, save and complete the task form', async () => {
```
Comment (DenysVuika) [inline e2e/process-services-cloud/form-field/dropdown-widget.e2e.ts:133]: should not be used

File: e2e/process-services/pages/task-filters-demo.page.ts [hunk 4]
```diff
@@ -43,7 +43,7 @@ export class TaskFiltersDemoPage {
         return new TaskFiltersPage(this.involvedTask);
     }
 
-    customTaskFilter(filterName: string): TaskFiltersPage {
+      customTaskFilter(filterName: string): TaskFiltersPage {
```
Comment (DenysVuika) [inline e2e/process-services/pages/task-filters-demo.page.ts:46]: incorrect formatting

File: e2e/process-services/tasks/task-filters-sorting.e2e.ts
Comment (DenysVuika) [inline e2e/process-services/tasks/task-filters-sorting.e2e.ts:1]: should this be back ported to ADF earlier?

File: e2e/search/components/search-number-range.e2e.ts
Comment (DenysVuika) [inline e2e/search/components/search-number-range.e2e.ts:1]: maybe needs to be back ported earlier to ADF

File: e2e/search/components/search-slider.e2e.ts
Comment (DenysVuika) [inline e2e/search/components/search-slider.e2e.ts:1]: can be back ported earlier

File: lib/cli/package.json [hunk 5]
```diff
@@ -4,12 +4,12 @@
   "version": "6.9.0",
   "author": "Hyland Software, Inc. and its affiliates",
   "bin": {
-    "adf-cli": "./bin/adf-cli",
-    "adf": "./bin/adf-cli"
+    "adf-cli": "bin/adf-cli",
```
Comment (DenysVuika) [inline lib/cli/package.json:7]: can be back ported to ADF earlier

File: lib/content-services/src/lib/category/categories-management/categories-management.component.html
Comment (DenysVuika) [inline lib/content-services/src/lib/category/categories-management/categories-management.component.html:1]: can be ported to ADF earlier

File: lib/content-services/src/lib/category/categories-management/categories-management.component.ts [hunk 6]
```diff
@@ -237,11 +229,13 @@ export class CategoriesManagementComponent implements OnInit, OnDestroy {
      * @param category - selection list change containing selected category
      */
     addCategoryToAssign(category: Category) {
-        const selectedCategory: Category = category;
-        this.categories.push(selectedCategory);
-        this._existingCategories.splice(this._existingCategories.indexOf(selectedCategory), 1);
-        this.categoryNameControl.updateValueAndValidity();
-        this.categoriesChange.emit(this.categories);
+        if (!(this.isCRUDMode || (!this.multiSelect && this.categories.length > 0))) {
```
Comment (DenysVuika) [inline lib/content-services/src/lib/category/categories-management/categories-management.component.ts:232]: should be backported earlier

File: lib/content-services/src/lib/common/services/people-content.service.spec.ts [hunk 7]
```diff
@@ -16,17 +16,23 @@
...
+import { EMPTY, of } from 'rxjs';
 
 describe('PeopleContentService', () => {
     let peopleContentService: PeopleContentService;
 
     beforeEach(() => {
         TestBed.configureTestingModule({
-            imports: [CoreTestingModule],
-            providers: [{ provide: AlfrescoApiService, useClass: AlfrescoApiServiceMock }]
+            imports: [HttpClientTestingModule],
```
Comment (DenysVuika) [inline lib/content-services/src/lib/common/services/people-content.service.spec.ts:30]: is there any reason for doing that during migration? the testing module was to save the code, not to have same stuff in each test

File: lib/content-services/src/lib/content-metadata/components/content-metadata-card/content-metadata-card.component.spec.ts [hunk 8]
```diff
@@ -34,15 +45,40 @@ describe('ContentMetadataCardComponent', () => {
...
+    let tagService: TagService = null;
+    let categoryService: CategoryService = null;
+    let propertyDescriptorsService: PropertyDescriptorsService = null;
 
     const getToggleEditButton = () => fixture.debugElement.query(By.css('[data-automation-id="meta-data-card-toggle-edit"]'));
 
     beforeEach(() => {
         TestBed.configureTestingModule({
-            imports: [ContentTestingModule]
+            imports: [
```
Comment (DenysVuika) [inline lib/content-services/src/lib/content-metadata/components/content-metadata-card/content-metadata-card.component.spec.ts:56]: this does not make any sense

File: lib/content-services/src/lib/dialogs/download-zip/download-zip.dialog.spec.ts [hunk 9]
```diff
@@ -37,10 +40,14 @@ describe('DownloadZipDialogComponent', () => {
 
     beforeEach(() => {
         TestBed.configureTestingModule({
-            imports: [CoreTestingModule],
```
Comment (DenysVuika) [inline lib/content-services/src/lib/dialogs/download-zip/download-zip.dialog.spec.ts:40]: This change is not required

File: lib/content-services/src/lib/dialogs/download-zip/download-zip.dialog.stories.ts [hunk 10]
```diff
@@ -15,30 +15,26 @@
...
-} from './mock/download-zip-service.mock';
-import { DownloadZipDialogModule } from './download-zip.dialog.module';
+import { AlfrescoApiServiceMock, ContentApiMock, DownloadZipMockService, NodesApiMock } from './mock/download-zip-service.mock';
 import { DownloadZipService } from './services/download-zip.service';
 import { ContentService } from '../../common/services/content.service';
 import { NodesApiService } from '../../common/services/nodes-api.service';
+import { MatDialogModule } from '@angular/material/dialog';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { importProvidersFrom } from '@angular/core';
+import { CoreStoryModule } from '../../../../../core/src/public-api';
```
Comment (DenysVuika) [inline lib/content-services/src/lib/dialogs/download-zip/download-zip.dialog.stories.ts:29]: something wrong with this import path, the story module should not be compiled together with the core lib

File: demo-shell/src/app.config.json [hunk 11]
```diff
@@ -622,7 +622,7 @@
             "unit": "Byte",
             "field": "cm:content.size",
             "min": 0,
-            "max": 18,
+            "max": 4000,
```
Comment (DenysVuika) [inline demo-shell/src/app.config.json:625]: is this change required?

File: demo-shell/src/app/app.module.ts [hunk 12]
```diff
@@ -73,7 +73,7 @@ import { FolderDirectiveModule } from './folder-directive';
         BrowserModule,
         environment.e2e ? NoopAnimationsModule : BrowserAnimationsModule,
         ReactiveFormsModule,
-        RouterModule.forRoot(appRoutes, { useHash: true, relativeLinkResolution: 'legacy' }),
+        RouterModule.forRoot(appRoutes, { useHash: true }),
```
Comment (DenysVuika) [inline demo-shell/src/app/app.module.ts:76]: should be ported to ADF earlier

File: e2e/process-services-cloud/form-field/visibility-condition-tabs.e2e.ts [hunk 13]
```diff
@@ -41,13 +46,13 @@ describe('Visibility conditions on tabs - cloud', () => {
     };
 
     const tab = {
-        tabWithFields: 'tabWithFields',
```
Comment (DenysVuika) [inline e2e/process-services-cloud/form-field/visibility-condition-tabs.e2e.ts:44]: what's the point of these changes?

File: lib/content-services/src/lib/aspect-list/aspect-list-dialog.component.spec.ts [hunk 14]
```diff
@@ -113,7 +114,7 @@ describe('AspectListDialogComponent', () => {
             excludedAspects: []
         };
         await TestBed.configureTestingModule({
-            imports: [ContentTestingModule, MatDialogModule],
+            imports: [HttpClientTestingModule, ContentTestingModule, MatDialogModule],
```
Comment (DenysVuika) [inline lib/content-services/src/lib/aspect-list/aspect-list-dialog.component.spec.ts:117]: Add the `HttpClientTestingModule` to the `ContentTestingModule` if it's not yet there, and that will reduce the code duplication

File: lib/content-services/src/lib/content-metadata/components/content-metadata-card/content-metadata-card.component.spec.ts (hunk 8 above)
Comment (VitoAlbano) [reply lib/content-services/src/lib/content-metadata/components/content-metadata-card/content-metadata-card.component.spec.ts:56]: It does actually, all that stuff that was embedded in the ContentTEstingModule was hiding the fact that this unit test was trying to do real REST calls, which is an issue that was at a certain point blocking the execution of the whole chain. 
As this was the only way to find out, it's working and it is stable now so i consider this an improvement actually, but feel free to change it as you wish after we merged the migration for ng15

File: lib/content-services/src/lib/common/services/people-content.service.spec.ts (hunk 7 above)
Comment (VitoAlbano) [reply lib/content-services/src/lib/common/services/people-content.service.spec.ts:30]: unit tests were unstable and trying to do actual REST calls , some of them were failing polluting the logs a lot as we discussed that's why some of them are changed
